shrinklvm.py

```diff
@@ -0,0 +1,1012 @@
+
+        for seg in lv.segs:
+            assert(seg.get_config('stripe_count') == '1')
+
+
+# A basic state machine for parsing the LVM text metadata format.
+
+class Enum:
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+
+    def __eq__(self, other):
+        return self.name == other.name and self.value == other.value
+
+
+class ConfigState(Enum):
+    INVALID = Enum('INVALID', 0)
+    TOP = Enum('TOP', 1)
+    IN_VG_I = Enum('IN_VG_I', 2)
+    IN_PV = Enum('IN_PV', 3)
+    IN_PV_I = Enum('IN_PV_I', 4)
+    IN_LV = Enum('IN_LV', 5)
+    IN_LV_I = Enum('IN_LV_I', 6)
+    IN_SEG = Enum('IN_SEG', 7)
+    IN_STRIPES = Enum('IN_STRIPES', 8)
+    MAX = Enum('MAX', 9)
+
+
+state_moves = {
+    '{': [ConfigState.INVALID] * ConfigState.MAX.value,
+    '[': [ConfigState.INVALID] * ConfigState.MAX.value,
+    ']': [ConfigState.INVALID] * ConfigState.MAX.value,
+    '}': [ConfigState.INVALID] * ConfigState.MAX.value,
+}
+
+state_moves['{'][ConfigState.TOP.value] = ConfigState.IN_VG_I
+state_moves['{'][ConfigState.IN_PV.value] = ConfigState.IN_PV_I
+state_moves['{'][ConfigState.IN_LV.value] = ConfigState.IN_LV_I
+state_moves['{'][ConfigState.IN_LV_I.value] = ConfigState.IN_SEG
+
+state_moves['}'][ConfigState.IN_VG_I.value] = ConfigState.TOP
+state_moves['}'][ConfigState.IN_PV.value] = ConfigState.IN_VG_I
+state_moves['}'][ConfigState.IN_PV_I.value] = ConfigState.IN_PV
+state_moves['}'][ConfigState.IN_LV.value] = ConfigState.IN_VG_I
+state_moves['}'][ConfigState.IN_LV_I.value] = ConfigState.IN_LV
+state_moves['}'][ConfigState.IN_SEG.value] = ConfigState.IN_LV_I
+
+state_moves['['][ConfigState.IN_SEG.value] = ConfigState.IN_STRIPES
+state_moves[']'][ConfigState.IN_STRIPES.value] = ConfigState.IN_SEG
+
+
+def next_state(state, char):
+    return state_moves[char][state.value]
+
+
+def action(lvm, state, l):
+    if state == ConfigState.IN_STRIPES:
+        lvm.vgs[-1].lvs[-1].segs[-1].stripes.append(l)
+    else:
+        items = l.split('=')
+        assert(len(items) == 2)
+        key, val = (i.strip() for i in items)
+
+        if state == ConfigState.TOP:
+            lvm.config.append((key, val))
+        elif state == ConfigState.IN_VG_I:
+            lvm.vgs[-1].config.append((key, val))
+        elif state == ConfigState.IN_PV_I:
+            lvm.vgs[-1].pvs[-1].config.append((key, val))
+        elif state == ConfigState.IN_LV_I:
+            lvm.vgs[-1].lvs[-1].config.append((key, val))
+        elif state == ConfigState.IN_SEG:
+            lvm.vgs[-1].lvs[-1].segs[-1].config.append((key, val))
+        else:
+            assert(False)
+
+
+def parse_config(lvm):
+    '''Parse an LVM config
+
+    Given a text config, parse it into VGs, PVs, etc.
+    '''
+
+    state = ConfigState.TOP
+    for l in lvm.text_config.split('\n'):
+        l = l.strip()
+        if not l:
+            continue
+
+        l = l.rsplit('#')[0].strip()
+        if not l:
+            continue
+
+        before_state = state
```

Comment:
This variable is never used

Reply:
That was some left over debugging. Removed.